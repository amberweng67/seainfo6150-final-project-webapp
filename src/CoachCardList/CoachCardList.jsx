import React from "react";
import styles from './CoachCardList.module.css';
import CoachCard from '../CoachCard/CoachCard';

const CoachCardList = ({ cardList }) => {
  if (!cardList) {
    return null;
  }
  if (cardList.length === 0) {
    return 'There is no coach data right now';
  }

  return (
    <div className={styles.listContainer}>
      {cardList.map(card => <CoachCard key={card.id} card={card} />)}
    </div>
  );
};

export default CoachCardList;
