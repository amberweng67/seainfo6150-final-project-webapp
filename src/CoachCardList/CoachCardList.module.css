.listContainer {
  display: flex;
}

.listContainer > *:not(:last-child) {
  margin-right: 16px;
}