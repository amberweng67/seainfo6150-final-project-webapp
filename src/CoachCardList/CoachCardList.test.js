import React from "react";
import CoachCardList from "./CoachCardList.jsx";

const CARD_LIST = [
  {
     "id": "a45e2d9c-cc14-5d54-844f-c6da44f94d0f",
     "name": "Grace Lynch",
     "title": "Senior Product Manager",
     "company": "She inc",
     "intro": "Taught block rear previous pain though happily war wealth view image hay dull proud health eventually building food citizen necessary angry circus total middle"
   },
   {
     "id": "9e4c2e5c-3856-5530-90af-2d603d39c271",
     "name": "Minerva Waters",
     "title": "Product Manager",
     "company": "Stopped inc",
     "intro": "Travel enough each send income grew zoo duck complete sit grain colony after distance toward heart involved date accurate cat vessels carefully whale shoe"
   }
];

it("renders the coach card list component correctly", () => {
  const { container } = render(<CoachCardList cardList={CARD_LIST}/>);
  expect(container).toMatchSnapshot();
});
