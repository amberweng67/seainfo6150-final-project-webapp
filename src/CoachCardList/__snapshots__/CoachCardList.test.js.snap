// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the coach card list component correctly 1`] = `
<div>
  <div
    class="listContainer"
  >
    <a
      class="link"
      href="/coaches/a45e2d9c-cc14-5d54-844f-c6da44f94d0f"
    >
      <div
        class="cardContainer"
      >
        <div>
          Grace Lynch
        </div>
        <div>
          Senior Product Manager
        </div>
        <div>
          She inc
        </div>
      </div>
    </a>
    <a
      class="link"
      href="/coaches/9e4c2e5c-3856-5530-90af-2d603d39c271"
    >
      <div
        class="cardContainer"
      >
        <div>
          Minerva Waters
        </div>
        <div>
          Product Manager
        </div>
        <div>
          Stopped inc
        </div>
      </div>
    </a>
  </div>
</div>
`;
